lib/diagnostic.js

```diff
@@ -8,40 +8,71 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
+  let lowNum = arguments[0];
+  for (let i = 0; i < arguments.length; i++) {
+    if (lowNum > arguments[i]) {
+      lowNum = arguments[i];
+    }
+  }
 
+  return lowNum;
 };
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
+  let nums = [];
+  let result = 0;
+  for (let i = 0; i < nums.length; i++) {
+    result += nums[i];
+  }
 
+  return result;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
-const newArray = function newArray(/* length, defaultValue */) {
+const newArray = function newArray(length, defaultValue) {
+  let result = [];
+  for (let i = 0; i < length; i++) {
+    result.push(defaultValue);
+  }
 
+  return result;
 };
 
 // Question 4
 // Add code to newArray so that if defaultValue is a
 // function then the elements in the array will be set
 // to the result of invoking defaultValue with the
 // element's index as an argument
+const defaultValue = function () {
+  return 'element';
+};
+
+newArray(5, defaultValue);
 
 // Questions 5
 // Write a function that adds or replaces a method on an object
 // using the given propertyName
 // If the argument obj is unset, create a new object
 // The function you write should return
 // the object passed in as an argument or the object created
-const addMethod = function addMethod(/* propertyName, method, obj */) {
 
+let hat = {
+  name: 'hat',
+  color: 'orange',
 };
 
+const addMethod = function addMethod(propertyName, method, obj) {
+  propertyName[method] = obj;
```

Comment:
This line breaks the test because it's trying to set a property called ```method``` onto the string ```propertyName``` with ```obj``` as the value.  The correct line is ```obj[propertyName] = method;``` if ```obj``` is an object.